cuppa/src/test/java/org/forgerock/cuppa/ModelTests.java

```diff
@@ -50,8 +50,10 @@ public void testBlockShouldBeImmutable() {
                 HookFunction.identity())))
                 .isExactlyInstanceOf(UnsupportedOperationException.class);
 
-        assertThatThrownBy(() -> testBlock.options.set(new TestOption("a")))
-                .isExactlyInstanceOf(UnsupportedOperationException.class);
+        Options options = testBlock.options;
+        testBlock.options.set(new TestOption("a"));
```

Comment:
Indeed it does. Then I'd just delete the new assertions from here. This test can safely assume that `Options` is immutable.